addons/l10n_in_withholding/wizard/l10n_in_account_withhold_wizard.py

```diff
@@ -0,0 +1,303 @@
+            'ref': self.reference,
+            'l10n_in_is_withholding': True,
+            'l10n_in_withholding_ref_move_id': self.related_move_id.id or self.related_payment_id.move_id.id,
+        }
+        return vals
+
+    def _prepare_withhold_move_lines(self, withholding_account_id):
+        """
+        Prepare the move lines for the withhold entry
+        """
+        def append_vals(quantity, price_unit, debit, credit, account_id, tax_ids):
+            return {
+                'quantity': quantity,
+                'price_unit': price_unit,
+                'debit': debit,
+                'credit': credit,
+                'account_id': account_id.id,
+                'tax_ids': tax_ids,
+            }
+
+        vals = []
+        total_amount = 0
+        total_tax = 0
+
+        related_move_id = self.related_move_id or self.related_payment_id.move_id
+
+        if self.withhold_type in ('in_withhold', 'in_refund_withhold'):
+            partner_account = related_move_id.partner_id.property_account_payable_id
+        else:
+            partner_account = related_move_id.partner_id.property_account_receivable_id
+
+        # Create move lines for each withhold line with the withholding tax and the base amount
+        for line in self.withhold_line_ids:
+            debit = line.base if self.withhold_type in ('in_withhold', 'out_refund_withhold') else 0.0
+            credit = 0.0 if self.withhold_type in ('in_withhold', 'out_refund_withhold') else line.base
+            vals.append(append_vals(1.0, line.base, debit, credit, withholding_account_id, [Command.set(line.tax_id.ids)]))
+            total_amount += line.base
+            total_tax += line.amount
+
+        # Create move line for the sum of all withhold lines (total amount)
+        debit = 0.0 if self.withhold_type in ('in_withhold', 'out_refund_withhold') else total_amount
+        credit = total_amount if self.withhold_type in ('in_withhold', 'out_refund_withhold') else 0.0
+        vals.append(append_vals(1.0, total_amount, debit, credit, withholding_account_id, False))
+
+        # Create move line for the sum of all withhold taxes (total tax)
+        debit = total_tax if self.withhold_type in ('in_withhold', 'out_refund_withhold') else 0.0
+        credit = 0.0 if self.withhold_type in ('in_withhold', 'out_refund_withhold') else total_tax
+        vals.append(append_vals(1.0, total_tax, debit, credit, partner_account, False))
+
+        return vals
+
+    def _validate_withhold_data_on_post(self, withholding_account_id):
+        if not withholding_account_id:
+            raise UserError(_("Please configure withholding account from settings"))
+        if not self.withhold_line_ids:
+            raise ValidationError(_("You must input at least one withhold line"))
+
+
+class L10nInAccountWithholdLine(models.TransientModel):
+    _name = 'l10n_in.account.withhold.line'
+    _description = "Withhold Wizard Lines"
+
+    base = fields.Monetary(string="Base")
```

Comment:
It is not necessary to deduct TDS on the entire invoice; TDS can be deducted for a partial invoice. Therefore, we decided it would be good for the user to input the value.

Reply:
Yes, but by default we can put the entire invoice amount and then the user can still change it.

Reply:
We intensely do not compute it.
Because it's the maximum chance that the user makes a mistake in quick action.
There is frequently a case where the user needs to modify amount.
like Advance TDS, different types of services that attract different amounts.

Reply:
Also for simplicity.
if the computed amount, we need to take care of the advanced TDS reconciled and the previous TDS reconciled.
we need this ASAP so we make it as simple as possible.